python/mlc_chat/nn/kv_cache.py

```diff
@@ -379,7 +381,7 @@ def _var(dtype):
     return T.alloc_buffer((1,), dtype, scope="local")
 
 
-def _attention_prefill(h_kv, h_q, d, dtype):
+def _attention_prefill(h_kv, h_q, d, dtype, target: Target):  # pylint: disable=unused-argument
```

Comment:
@CharlieFRuan These functions now become target dependent. Would you like to follow up with the WebGPU support? I assume that would be a matter of detecting target and mutate tile sizes accordingly.

Reply:
Will do, thank you!

Reply:
Should we add `target` to other kernels as well? e.g. for `tir_rotary()` I had to manually change the layout as well when compiling for WebGPUs: https://github.com/CharlieFRuan/mlc-llm/commit/7bf23e1f19b3bcf3aa68d30b0ba5171b313198f2#diff-2dcbda4722cf510f0e80ddf1d4df62f77108e56c192bebcbaa3a51308763cf0fR322.

Reply:
Thanks for the catch! Just added.